odbc_fdw.c

```diff
@@ -1358,8 +1433,87 @@ odbcIterateForeignScan(ForeignScanState *node)
 			   SQL_C_TYPE_DATE/SQL_C_TYPE_TIME/SQL_C_TYPE_TIMESTAMP.
 			   And finally, SQL_C_NUMERIC and SQL_C_GUID could also be used.
 			*/
+			buf[0] = 0;
+			buf_used = 0;
 			ret = SQLGetData(stmt, i, SQL_C_CHAR,
 							 buf, sizeof(char) * (col_size+1), &indicator);
+			buf_used = indicator;
+
+			if (ret == SQL_SUCCESS_WITH_INFO)
+			{
+				SQLCHAR sqlstate[5];
+				SQLGetDiagRec(SQL_HANDLE_STMT, stmt, 1, sqlstate, NULL, NULL, 0, NULL);
+				if (strcmp((char*)sqlstate, "01S07") == 0)
```

Comment:
It's the ODBC SQLSTATE for "Fractional truncation".

I'll define a constant to make it more clear.